docs/guides/typescript_project.mdx

```diff
@@ -44,93 +35,111 @@ To use TypeScript in our projects, we'll need the following prerequisites.
     npm install --dev @types/node
     ```
 
-4. TypeScript configuration file allowing `tsc` to understand the project layout and
-   the features used in the project:
+4. TypeScript configuration file allowing `tsc` to understand the project layout and the features used in the project:
 
-    ```json
+    > We are extending the [`@apify/tsconfig`](https://github.com/apify/apify-tsconfig), it contains [the set of rules](https://github.com/apify/apify-tsconfig/blob/main/tsconfig.json) we believe are worth following.
+
+    > To be able to use feature called [Top level await](https://blog.saeloun.com/2021/11/25/ecmascript-top-level-await.html), we will need to set the `module` and `target` compiler options to `ES2022` or above. This will make the project compile to [ECMAScript Modules](https://nodejs.org/api/esm.html).
+
+    ```json title="tsconfig.json"
     {
+        "extends": "@apify/tsconfig",
         "compilerOptions": {
-            "target": "es2019",
-            "module": "commonjs",
-            "moduleResolution": "node",
-            "strict": true,
-            "noImplicitAny": false,
-            "strictNullChecks": false,
-            "lib": [
-                "DOM",
-                "DOM.Iterable",
-                "ES2015",
-                "ES2016",
-                "ES2018",
-                "ES2019.Object",
-                "ES2018.AsyncIterable",
-                "ES2020.String",
-                "ES2019.Array"
-            ],
-            "rootDir": "src/",
-            "outDir": "build/"
+            "module": "ES2022",
+            "target": "ES2022",
+            "outDir": "dist",
+            "lib": ["DOM"]
```

Comment:
Do we want to document the usage of the DOM types too? It's *technically* correct for browser scrapers, but not exactly needed for cheerio for example